client/blocks/login/login-header.tsx

```diff
@@ -0,0 +1,532 @@
+
+	if ( twoFactorAuthType === 'push' ) {
+		headerText = translate( 'Continue with the Jetpack app' );
+	} else if ( twoFactorAuthType === 'backup' ) {
+		headerText = translate( 'Continue with a backup code' );
+	}
+
+	if ( isManualRenewalImmediateLoginAttempt ) {
+		headerText = translate( 'Log in to update your payment details and renew your subscription' );
+	}
+
+	if ( twoStepNonce ) {
+		headerText = translate( 'Two-Step Authentication' );
+	}
+
+	if ( socialConnect ) {
+		headerText = translate( 'Connect your %(service)s account', {
+			args: {
+				service: capitalize( linkingSocialService ),
+			},
+		} );
+	}
+
+	if ( action === 'lostpassword' ) {
+		headerText = translate( 'Forgot your password?' );
+	}
+
+	if ( oauth2Client ) {
+		if ( isWCCOM ) {
+			if ( wccomFrom === 'cart' ) {
+				headerText = translate( 'Log in with a WordPress.com account' );
+			} else if ( twoFactorEnabled ) {
+				headerText = translate( 'Authenticate your login' );
+			} else if ( currentQuery.lostpassword_flow ) {
+				headerText = translate( 'Log in to your account' );
+			} else if ( showContinueAsUser ) {
+				headerText = (
+					<>
+						{ wccomFrom === 'nux'
+							? translate( 'Get started in minutes' )
+							: translate( 'Log in to your account' ) }
+					</>
+				);
+			}
+			headerText = translate( 'Log in to your account' );
+		}
+
+		if ( isJetpackCloudOAuth2Client( oauth2Client ) ) {
+			headerText = translate( 'Howdy! Log in to Jetpack.com with your WordPress.com account.' );
+		}
+
+		if ( isA4AOAuth2Client( oauth2Client ) ) {
+			headerText = translate(
+				'Howdy! Log in to Automattic for Agencies with your WordPress.com{{nbsp/}}account.',
+				{
+					components: { nbsp: <>&nbsp;</> },
+					comment: 'The {{nbsp/}} is a non-breaking space',
+				}
+			);
+		}
+
+		if ( isPartnerPortalOAuth2Client( oauth2Client ) ) {
+			if ( document.location.search?.includes( 'wpcloud' ) ) {
+				headerText = translate( 'Log in to WP Cloud with WordPress.com' );
+			} else if ( document.location.search?.includes( 'jetpack' ) ) {
+				headerText = translate(
+					'Howdy! Log into the Jetpack Partner Portal with your WordPress.com account.'
+				);
+			} else {
+				headerText = translate(
+					'Howdy! Log into the Automattic Partner Portal with your WordPress.com account.'
+				);
+			}
+		}
+
+		if ( isCrowdsignalOAuth2Client( oauth2Client ) ) {
+			headerText = translate( 'Sign in to %(clientTitle)s', {
+				args: {
+					clientTitle: oauth2Client.title,
+				},
+			} );
+		}
+
+		if ( isGravPoweredClient ) {
+			headerText = translate( 'Login to %(clientTitle)s', {
+				args: { clientTitle: oauth2Client.title },
+			} );
+		}
+
+		if ( isBlazeProOAuth2Client( oauth2Client ) ) {
+			headerText = translate( 'Log in to your Blaze Pro account' );
+		}
+	}
+
+	if ( isWooJPC ) {
+		const isLostPasswordFlow = currentQuery.lostpassword_flow === 'true';
+		switch ( true ) {
+			case isLostPasswordFlow:
+				headerText = translate( "You've got mail" );
```

Comment:
There are missing `break` statements in this `switch`. The `isLostPasswordFlow` condition falls through to the following condtions and the `headerText` is overwritten.

Personally I would rewrite these `switch ( true )` statements to a regular `if ... else if ... else`.